views/js/test/qtiRunner/modalFeedback/choiceSelector/test.js

```diff
@@ -0,0 +1,427 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ */
+define([
+    'jquery',
+    'lodash',
+    'taoQtiItem/qtiCreator/editor/response/choiceSelector'
+], function($, _, choiceSelector) {
+    'use strict';
+
+    QUnit.module('choiceSelector');
+
+
+    QUnit.test('module', 3, function(assert) {
+        assert.equal(typeof choiceSelector, 'function', "The choiceSelector module exposes a function");
+        assert.equal(typeof choiceSelector(), 'object', "The choiceSelector factory produces an object");
+        assert.notStrictEqual(choiceSelector(), choiceSelector(), "The choiceSelector factory provides a different object on each call");
+
+    });
+
+
+    var testReviewApi = [
+        { name : 'init', title : 'init' },
+        { name : 'destroy', title : 'destroy' },
+        { name : 'render', title : 'render' },
+        { name : 'show', title : 'show' },
+        { name : 'hide', title : 'hide' },
+        { name : 'enable', title : 'enable' },
+        { name : 'disable', title : 'disable' },
+        { name : 'is', title : 'is' },
+        { name : 'setState', title : 'setState' },
+        { name : 'getElement', title : 'getElement' },
+        { name : 'getContainer', title : 'getContainer' },
+        { name : 'getTemplate', title : 'getTemplate' },
+        { name : 'setTemplate', title : 'setTemplate' },
+        { name: 'getChoices', title: 'getChoices'}
+    ];
+
+    QUnit
+        .cases(testReviewApi)
+        .test('instance API ', function(data, assert) {
+            var instance = choiceSelector();
+            assert.equal(typeof instance[data.name], 'function', 'The choiceSelector instance exposes a "' + data.title + '" function');
+            instance.destroy();
+        });
+
+    //
```

Comment:
Please complete the unit tests for the init and destroy events for your component, multiple instanciation, and the api methods: getChoices, setChoices, resetChoices